### PR TITLE
feat(dual-list-selector): added drag/drop

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/4319

### CODE REVIEW

**Review by mcarrano — APPROVED** [review]
Looks great. Thanks @mcoker !

**Review by mattnolting — COMMENTED** [review]
Looks great! @mcarrano can you confirm that this text should be blue? Looks like a link to me.

Screen Shot 2021-09-08 at 11 54 08 AM

**Comment by mcarrano** [comment]
@mattnolting Yes, that is the styling currently used for selected items in the dual list.

File: src/patternfly/components/DualListSelector/examples/DualListSelector.md [hunk 1]
```diff
@@ -861,7 +994,9 @@ cssPrefix: pf-c-dual-list-selector
 | `.pf-c-dual-list-selector__controls-item` | `<div>` | Initiates the dual list selector controls item. **Required** |
 | `.pf-m-available` | `.pf-c-dual-list-selector__pane` | Defines a pane as the available list. |
 | `.pf-m-chosen` | `.pf-c-dual-list-selector__pane` | Defines a pane as the chosen list. |
-| `.pf-m-selected` | `.pf-c-dual-list-selector__item` | Modifies the menu item for the selected state. |
-| `.pf-m-check` | `.pf-c-dual-list-selector__item` | Indicates that an item is selectable with a checkbox. |
+| `.pf-m-drag-over` | `.pf-c-dual-list-selector__list` | Modifies the dual list selector list to indicate that a draggable item is being dragged over the list. |
+| `.pf-m-ghost-row` | `.pf-c-dual-list-selector__list-item-row` | Modifies the list item main to be a ghost row. |
+| `.pf-m-selected` | `.pf-c-dual-list-selector__list-item-row` | Modifies the menu item for the selected state. |
```
Comment (srambach) [inline src/patternfly/components/DualListSelector/examples/DualListSelector.md:999]: Is "list item main" the same as "menu item" ?

Comment (mcoker) [reply src/patternfly/components/DualListSelector/examples/DualListSelector.md:999]: Nope - `.pf-m-selected` is marking the item in list-item-row as selected, but the modifier needs to be on list-item-row so the styling works.

**Review by mattnolting — APPROVED** [review]
Great work, LPTM!

**Pull request merged**